micrometer-core/src/main/java/io/micrometer/core/instrument/TracingTimer.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.Tracer;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+public class TracingTimer implements Timer{
+
+    static Builder builder(String name, Tracer tracer) {
+        return new Builder(name, tracer);
+    }
+
+    static class Builder extends Timer.Builder{
+        private final Tracer tracer;
+
+        Builder(String name, Tracer tracer) {
+            super(name);
+            this.tracer = tracer;
+        }
+
+        @Override
+        public Timer register(MeterRegistry registry) {
```

Comment:
any reason we can't return the more specific type `TracerTimer`?

Reply:
Good idea. I had a flashback of Java 4 not supporting covariant return types.